### PR TITLE
chore: fix links due to ts upgrade

### PR DESCRIPTION
**Pull request by williamw2**
# Pull Request

## 📖 Description

The Typescript version upgrade changes how it handles the @link JSDOC tag in comments causing them to be removed when the CEM Analyzer process the code. This results in the link URLs and text being rendered to markdown in such a way that they cannot be corrected into markdown links. This PR adds a CEM Analyzer plugin that checks for the existence of @link tags in the descriptions generated by "comment-parser" and retains the original text.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { getClassMemberDoc } from "@custom-elements-manifest/analyzer/src/utils/manifest-helpers.js";
+import { parse } from "comment-parser";
+
+/**
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs:4]: FYI fast-foundation is a module package now, so you should be able to just use the `.js` extension if that is preferred.

File: packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                if (!propertyDoc)
+                    return (
+                        context.dev &&
+                        console.warn(
+                            `[jsdoc-example]: Could not find property ${propertyName} of ${className} in module doc for path ${moduleDoc.path}`
+                        )
+                    );
+                fixLinks(node, propertyDoc, context, 1);
+            } else if (ts.isMethodDeclaration(node)) {
+                // TODO: handle object methods
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs:91]: Is this todo still open? Should it have a tracking issue?

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs (hunk 1 above)
Comment (williamw2) [reply packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs:4]: I'm going to need to update some of this once we upgrade the API-Extractor and I add support for @slot tags. I can change the file extensions then.

File: packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs (hunk 2 above)
Comment (williamw2) [reply packages/web-components/fast-foundation/custom-elements-manifest-plugins.mjs]: Ah, no. Was copied over from the example plug-in code I used. Removed.

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**